src/main/java/net/dv8tion/jda/internal/handle/ChannelUpdateHandler.java

```diff
@@ -226,6 +229,44 @@ private AbstractGuildChannelImpl<?> handleChannelTypeChange(AbstractGuildChannel
 
             return newsChannel;
         }
+        else if (newChannelType == ChannelType.MEDIA && channel.getType() == ChannelType.FORUM)
+        {
+            ForumChannel forumChannel = (ForumChannel) channel;
+            getJDA().getForumChannelsView().remove(forumChannel.getIdLong());
+            guild.getForumChannelsView().remove(forumChannel.getIdLong());
+
+            MediaChannelImpl mediaChannel = (MediaChannelImpl) builder.createMediaChannel(guild, content, guild.getIdLong());
+
+            getJDA().handleEvent(
+                new ChannelUpdateTypeEvent(
+                    getJDA(), responseNumber,
+                    mediaChannel, ChannelType.FORUM, ChannelType.MEDIA));
+        }
+        else
+        {
+            WebSocketClient.LOG.warn("Received unexpected channel type change {}->{}", channel.getType(), newChannelType);
+
+            // Attempt to split into delete/create events
+            WebSocketClient client = getJDA().getClient();
+
+            DataObject syntheticDelete = DataObject.empty()
+                .put("t", "CHANNEL_DELETE")
+                .put("s", responseNumber)
+                .put("op", 0)
+                .put("d", DataObject.empty()
+                    .put("type", channel.getType().getId())
+                    .put("guild_id", guild.getId())
+                    .put("id", channel.getIdLong()));
+            client.getHandler("CHANNEL_DELETE").handle(responseNumber, syntheticDelete);
```

Comment:
I'm not sure we should do this from the perspective of it could cause bots to do things they'd normal do on `CHANNEL_DELETE` but incorrectly, like removing data for the channel even though the channel still exists